### DISCUSSION TITLE
Property '_renderControls' does not exist on type 'Object'.

### DISCUSSION THREAD

**Discussion by errmayank** [fabricjs/fabric.js#8546] [Q&A] [answered]
I'm using fabric `v5.3.0` and I'm trying to highlight objects on hover but I'm getting this typescript error that `_renderControls` does not exist

```ts
canvas.on('mouse:over', function (e) {
  if (e.target) {
    if (e.target === canvas.getActiveObject()) return;

    e.target.set('borderScaleFactor', 2);
    e.target._renderControls(canvas.getContext(), {
      hasControls: false,
    });
  }
});
```

Screenshot:

<img width="689" alt="Screenshot 2022-12-30 at 23 50 18" src="https://user-images.githubusercontent.com/64462998/210101199-2096fb59-3a76-4372-aef2-a36b60fdf10c.png">

It works but I'm just wondering if this method is going to be deprecated?

**Comment by ShaMan123** [comment, marked as answer]
I am assuming you are using definitely typed fabric (`@types/dabric`) which is stale, inaccurate and misleading.
Dump that and wait a week or so for us to publish v6 beta in typescript

**Reply by errmayank** [reply]
Ah I see, thanks!
